src/Admin/Redirect.php

```diff
@@ -81,62 +85,90 @@ public function activate(): void {
 	/**
 	 * Checks if merchant should be redirected to the onboarding page if it is not.
 	 *
-	 * @return bool True if the redirection should have happened
+	 * @return void
 	 */
-	protected function maybe_redirect_to_onboarding(): bool {
+	protected function maybe_redirect(): void {
 		if ( $this->wp->wp_doing_ajax() ) {
-			return false;
+			return;
 		}
 
-		// If we have redirected before do not attempt to redirect again.
-		if ( 'yes' !== $this->options->get( self::OPTION ) ) {
-			return false;
+		// Maybe redirect to onboarding after activation
+		if ( 'yes' === $this->options->get( self::OPTION ) ) {
+			$this->maybe_redirect_after_activation();
+			return;
+		}
+
+		// If setup ISNT complete then redirect from dashboard to onboarding
+		if ( ! $this->merchant_center->is_setup_complete() && $this->is_current_wc_admin_page( self::PATHS['dashboard'] ) ) {
+			$this->redirect_to( self::PATHS['get_started'] );
+		}
+
+		// If setup IS complete then redirect from onboarding to dashboard
+		if ( $this->merchant_center->is_setup_complete() && $this->is_current_wc_admin_page( self::PATHS['get_started'] ) ) {
+			$this->redirect_to( self::PATHS['dashboard'] );
 		}
+	}
 
+	/**
+	 * Checks if merchant should be redirected to the onboarding page after extension activation.
+	 *
+	 * @return bool True if the redirection should have happened
+	 */
+	protected function maybe_redirect_after_activation(): bool {
 		// Do not redirect if setup is already complete
 		if ( $this->merchant_center->is_setup_complete() ) {
 			$this->options->update( self::OPTION, 'no' );
 			return false;
 		}
 
 		// if we are on the get started page don't redirect again
-		if ( $this->is_onboarding_page() && 'yes' === $this->options->get( self::OPTION, 'yes' ) ) {
+		if ( $this->is_current_wc_admin_page( self::PATHS['get_started'] ) && 'yes' === $this->options->get( self::OPTION, 'yes' ) ) {
```

Comment:
💅  I guess` 'yes' === $this->options->get( self::OPTION, 'yes' )` is not necessary since at this point the option is always YES.